Sources/Subscription/SubscriptionURL.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  SubscriptionURL.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+// MARK: - URLs, ex URL+Subscription
+
+public enum SubscriptionURL {
+    case baseURL
+    case purchase
+    case faq
+    case activateViaEmail
+    case addEmail
+    case manageEmail
+    case activateSuccess
+    case addEmailToSubscriptionSuccess
+    case addEmailToSubscriptionOTP
+    case manageSubscriptionsInAppStore
+    case identityTheftRestoration
+
+    // swiftlint:disable:next cyclomatic_complexity
+    public func subscriptionURL(environment: SubscriptionEnvironment.ServiceEnvironment) -> URL {
```

Comment:
Note for the future:
Going through the client PR I see this function's usage as a bit inconvenient. In my old PR I went with the route of having an URLprovider in the manager that was initialised with the current environment. Hence you just requested specific URL (by enum) from the provider. This way we could have simpler calls and instead of:
- `let url = SubscriptionURL.identityTheftRestoration.subscriptionURL(environment: Application.appDelegate.subscriptionManager.currentEnvironment.serviceEnvironment)`
there would be:
- `let url = Application.appDelegate.subscriptionManager.urlProvider.url(for: .identityTheftRestoration)`

Reply:
Agree, I was already working on this, replaced with a simpler `subscriptionManager.url(for: .baseURL)`